File: LATIHAN2/PYTHON/Mahasiswa.py
# load library
from SivitasAkademik import SivitasAkademik

# inheritance fr SivitasAkademik


class Mahasiswa(SivitasAkademik):
    # private karena tidak perlu diwariskan lagi
    def __init__(self, NIM, nama, jenis_kelamin, fakultas, prodi, asal_universitas, email_edu):
        # ambil constructor dr parent
        super().__init__("", nama, jenis_kelamin, asal_universitas, email_edu)
        self._NIM = NIM
        self._fakultas = fakultas
        self._prodi = prodi

    # setter getter
    def setNIM(self, NIM):
        self._NIM = NIM

    def getNIM(self):
        return self._NIM

    def setFakultas(self, fakultas):
        self._fakultas = fakultas

    def getFakultas(self):
        return self._fakultas

    def setProdi(self, prodi):
        self._prodi = prodi

    def getProdi(self):
        return self._prodi

File: LATIHAN2/C++/fix/Mahasiswa.cpp
//load library
#include <string>
#include "SivitasAkademik.cpp"
using namespace std;

//inheritance fr sivitasakademik
class Mahasiswa : public SivitasAkademik {
//private soalnya anak gk anakan lagi
private:
    string NIM;
    string fakultas;
    string prodi;

//constructor
public:
    Mahasiswa(string _NIM, string _nama, string _jenis_kelamin, string _fakultas, string _prodi, string _asal_universitas, string _email_edu) : NIM(_NIM), fakultas(_fakultas), prodi(_prodi), SivitasAkademik("", _nama, _jenis_kelamin, _asal_universitas, _email_edu) {}

    //setter getter
    void setNIM(string _NIM) {
        NIM = _NIM;
    }

    string getNIM() {
        return NIM;
    }

    void setFakultas(string _fakultas) {
        fakultas = _fakultas;
    }

    string getFakultas() {
        return fakultas;
    }

    void setProdi(string _prodi) {
        prodi = _prodi;
    }

    string getProdi() {
        return prodi;
    }
};
